src/ConnectionPool.jl

```diff
@@ -257,8 +279,8 @@ function IOExtras.closewrite(t::Transaction)
 
     t.c.writebusy = false
     t.c.writecount += 1                           ;@debug 2 "🗣  Write done: $t"
-    notify(t.c.writedone)
-    notify(poolcondition[Threads.threadid()])
+    _notify(t.c.writedone)
+    _notify(poolcondition[Threads.threadid()])
 
     @ensure !iswritable(t)
     return
```

Comment:
Same here; it looks to me like this whole section should be `lock`ed, not just within the `_notify` function itself.